### PR TITLE
Switch to callback(err, content) convention

### LINKED ISSUE #42: callback(success, res or err) => callback(err, res or undefined)
## tl;dr;
```js
var client = algoliasearch({
  applicationID: '',
  apiKey: ''
});

var index = client.initIndex('indexName');

index.search('YAW callback', function searchDone(err, res) {
  if (err) {
    // do something with err, it's an instance of `Error`
    return;
  }

  // everything is cool, use response
  console.log(res.hits);
})
```

## Boring stuff
The current callback convention for our JavaScript client is `index.search(callback)` where callback will be called with `callback(success, res or err)`.

A popular way to handle asynchronous JavaScript is to use the `callback(err or null, data or undefined)` convention.

Notable projects using or leveraging this callback convention:
- the whole [Node.js api](http://nodejs.org/api/)
- [caolan/async](https://github.com/caolan/async)
- [request/request](https://github.com/request/request#requestoptions-callback)
- most of the npm modules doing asyncrhonous stuff will use this convention

In the days of ["isomorphic JavaScript"](http://isomorphic.net/), we must stick with a well used convention to not fall into the "guys using fancy callback pattern" basket.

## TODO
- [x] move callback convention to `callback(err, res)`
- [x] note breaking changes in the issue if any

## References
- http://blog.izs.me/post/59142742143/designing-apis-for-asynchrony
- http://caolanmcmahon.com/posts/nodejs_style_and_structure/
- http://nodeguide.com/style.html#callbacks

Linked with algolia/algoliasearch-client-node#18.

### PR DESCRIPTION
**Pull request by vvo**
**BREAKING CHANGE**, all api methods now use the (err, content) convention

It means that instead of doing:

```js
  index.search(function found(success, content) {
    if (!success) {
      console.log('Error was:', content);
      return;
    }

    console.log('OK!', content);
  });
```

You must do:

```js
  index.search(function found(err, content) {
    if (err) {
      console.log('Error was:', err);
      return;
    }

    console.log('OK!', content);
  });
```

This is in par with most asynchronous APIS accepted/de facto conventions.
It helps being compatible with control flow libraries like `async` and
helps reduce WTF moments of Node.JS developers used to cb(err, content)

fixes #42

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -30,6 +30,32 @@ UNRELEASED
...
+      You must do:
+        ```js
+          index.search(function found(success, content) {
+            if (!success) {
+              console.log('Error was:', content);
+              return;
+            }
+
+            console.log('OK!', content);
+          });
```
Comment (redox) [inline ChangeLog]: Hmmmm, copy/paste failed?

Comment (vvo) [inline ChangeLog]: oh yeah :)

**Comment by vvo** [comment]
@redox @bobylito ready to merge, tests ok

File: .gitattributes [hunk 2]
```diff
@@ -1,6 +1,8 @@
...
- dist/algoliasearch.jquery.js -diff
- dist/algoliasearch.jquery.min.js -diff
+dist/algoliasearch.js -diff
+dist/algoliasearch.min.js -diff
+dist/algoliasearch.angular.js -diff
+dist/algoliasearch.angular.min.js -diff
+dist/algoliasearch.jquery.js -diff
+dist/algoliasearch.jquery.min.js -diff
+test/vendor/angular.min.js -diff
+test/vendor/jquery.min.js -diff
```
Comment (vvo) [inline .gitattributes]: I added local versions of jquery and angular to ease local testing

File: package.json [hunk 3]
```diff
@@ -54,7 +56,7 @@
     "build": "PACKAGE_VERSION=$(json -f package.json version) sh build.sh",
     "test": "zuul --phantom --ui tape test/run.js | tap-growl | tap-spec && npm run lint",
     "test-ci": "DEBUG=zuul* zuul --tunnel ngrok test/run.js && npm run lint",
-    "dev": "DEBUG=zuul* zuul --local 8080 --ui tape test/run.js",
+    "dev": "DEBUG=zuul* zuul --no-coverage --local 8080 --ui tape test/run.js",
```
Comment (vvo) [inline package.json]: coverage tab currently breaks sourcemaps, not so used atm

File: .gitattributes (hunk 2 above)
Comment (redox) [inline .gitattributes]: Those libraries are available on NPM, would a devDependency work?

Comment (vvo) [inline .gitattributes]: Yes I will investigate this option tomorrow morning, might be ok.

Comment (vvo) [inline .gitattributes]: @redox done using dev dependencies, thanks

**Pull request merged**